fpdf/fpdf.py

```diff
@@ -1066,7 +1066,14 @@ def add_font(self, family, style="", fname=None, uni=False):
                 unifilename = cache_dir / f"{ttffilename.stem}.pkl"
 
             # include numbers in the subset! (if alias present)
-            sbarr = list(range(57 if self.str_alias_nb_pages else 32))
+            # ensure that alias is mapped 1-by-1 additionally (must be replaceable)
+            if self.str_alias_nb_pages:
+                subset_max = 57
+                for char in self.str_alias_nb_pages:
+                    subset_max = max(subset_max, ord(char))
```

Comment:
Why this change?
In practice, `sbarr` wille become `list(range(125))` if `str_alias_nb_pages` is set.

I still wonder where this value of `57` is coming from...

Also, it terms of code, this would be shorter:
```python
subset_max = max(map(ord, self.str_alias_nb_pages))
```

Reply:
Substituting page numbers became more complicated. When `_substitute_page_number()` calls `str.replace()`, it is no longer known which font/mapping was used. For example, if a dedicated font is used for the page numbers, _{nb}_ would become _\x01\x02\x03\x04_. The same string is created if another text starts with four different characters (which is likely), which would be replaced as well then. The simplest solution to prevent conflicts is that all characters of the `str_alias_nb_pages` and its resulting digits always get the same mapping.

To avoid having to embed all 125 characters, `font["subset"]` should be of another type. I guess `dict.get()` is also faster than `list.index()` 😉 I'll give it a try.

Thanks for your feedback!

Cheers,
Moe